lib/house.rb

```diff
@@ -0,0 +1,42 @@
+class House
+  attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+    @above_market_average = false
+    @sum = 0
+    @area_room = []
+    @detail_hsh = {}
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    @above_market_average
+  end
+
+  def rooms_from_category(category)
+    @rooms.map do |room|
+      if room.category == category
+        room
+      end
+    end.compact
+  end
+
+  def area
+    @rooms.map do |room|
+      room.length * room.width.to_i
+      # require "pry"; binding.pry
+    end
+  end
+
+  def details
+    @detail_hsh  = {
+                      "price" => "$400000",
+                      "address" => "123 sugar lane"}
```

Comment:
You don't need the `@detail_hsh` here. You can just have this method return the has you want it to have.  Also, this is hardcoded--the way you have this written, every house would have a price of $400000 and an address of "123 sugar lane" for its details, which probably shouldn't be the case. 

```
def details
   info = {}
   info["price"] = "$" + @price.to_i
   info["address] = @address 
end 
```